src/openrct2/peep/Peep.cpp

```diff
@@ -1785,6 +1787,22 @@ Peep* Peep::Generate(const CoordsXYZ& coords)
 
     increment_guests_heading_for_park();
 
+#ifdef ENABLE_SCRIPTING
+    auto& hookEngine = OpenRCT2::GetContext()->GetScriptEngine().GetHookEngine();
+    if (hookEngine.HasSubscriptions(OpenRCT2::Scripting::HOOK_TYPE::GUEST_GENERATION))
+    {
+        auto ctx = OpenRCT2::GetContext()->GetScriptEngine().GetContext();
+
+        // Create event args object
+        auto obj = OpenRCT2::Scripting::DukObject(ctx);
+        obj.Set("id", peep->sprite_index);
+
+        // Call the subscriptions
+        auto e = obj.Take();
+        hookEngine.Call(OpenRCT2::Scripting::HOOK_TYPE::GUEST_GENERATION, e, true);
```

Comment:
guess this can be left as false as you don't need to modify the guest

Reply:
Game state is mutable because this is a non-client specific event.